### ISSUE TITLE
Partition column missing when reading parquet from Google Cloud Storage

### ISSUE THREAD

**Issue by toby-coleman** [dask/dask#7509]
**What happened**: When reading a partitioned parquet dataframe from Google Cloud Storage, the partition column is missing from the dataframe.

**What you expected to happen**: Expected a column in the dataframe returned by `.read_parquet` to contain partition.

**Minimal Complete Verifiable Example**:

See [this Colab notebook](https://colab.research.google.com/drive/1uCALWyrH1bYvVFk25frfbrAvARqn4KP3?usp=sharing)

```python
import dask.dataframe as dd
import dask
import pandas as pd

# Sample dataframe
dts = pd.date_range('2000-01-01', '2020-01-01', freq='1M')
df = pd.DataFrame(
    {'time': dts, 'value': list(range(len(dts))), 'year': dts.year}
).set_index('time')
print(df.head())

ddf = dd.from_pandas(df, npartitions=1)

# Write partitioned dataframe to local storage
ddf.to_parquet('/tmp/dask-test', engine='pyarrow', partition_on='year')
# Read back
ddf_read = dd.read_parquet('/tmp/dask-test', engine='pyarrow')
print(ddf_read.head())
# Dataframe contains a 'year' column

token = ... # Authenticate to GCP here

# Write partitioned dataframe to GCP
ddf.to_parquet('gs://my-bucket/dask-test', engine='pyarrow', partition_on='year', storage_options={'token': token})
# Read back
ddf_read = dd.read_parquet('gs://my-bucket/dask-test', engine='pyarrow', storage_options={'token': token})
ddf_read.head()
# Dataframe missing the 'year' column
```

**Anything else we need to know?**:

**Environment**:

- Dask version: 2021.04.0
- Python version: 3.7.10
- Operating System: Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
- Install method (conda, pip, source): pip
- Pyarrow version: 3.0.0
- Gcsfs version: 0.7.2

**Comment by jsignell** [comment]
I would recommend that you try an older version of pyarrow, but pinging @rjzamora in case you have a sense of what's going on. 

In the meantime @toby-coleman if you can create a minimal reproducible example that'd be very helpful.

**Comment by toby-coleman** [comment]
Looks like the problem is there on `pyarrow==2.0.0`. @jsignell does the example above/Colab notebook not work as a minimal reproducible example - all that I left out was my GCP token?

Having said that looks like a new version of `fsspec` has just been released, which breaks something else... so it needs to be run with `fsspec<0.9.0`.

**Comment by jsignell** [comment]
Sorry I had missed that this only happens on GCP. Can you dump an `ls` of the files in `gs://my-bucket/dask-test`?

**Comment by toby-coleman** [comment]
Just tried it again on the newly released `gcsfs==0.8.0` and now it works - the partition column gets returned when calling `.read_parquet` dataframe. Looks like the issue can be closed 😀

**Comment by rjzamora** [comment]
>Just tried it again on the newly released gcsfs==0.8.0 and now it works - the partition column gets returned when calling .read_parquet dataframe. Looks like the issue can be closed 😀

Cool, I just updated my env to pyarrow-3 to help here, and was about to second Julia's request for a file listing =)

**Comment by rjzamora** [comment]
Thanks for the update @toby-coleman - Feel free to re-open this if the problem shows up again with gcsfs>=0.8.0
